analysers/analyser_osmosis_highway_tunnel_bridge.py

```diff
@@ -117,16 +166,40 @@ def __init__(self, config, logger = None):
         #    title = T_('Missing maxheight tag'))
         #self.classs_change[3] = self.def_class(item = 7130, level = 3, tags = ['tag', 'highway', 'layer', "fix:imagery"],
         #    title = T_('Missing layer tag around bridge'))
+        self.classs_change[4] = self.def_class(item = 7012, level = 3, tags = ['highway', 'fix:survey', 'fix:imagery'],
+            title = T_('Bridge connected to non-bridge highway'),
+            detail = T_(
+'''A bridge or tunnel is usually not connected to regular highways except at the end points.'''),
+            fix = T_(
+'''Disconnect the bridge or tunnel from the highway, or add missing bridge or tunnel tags.
+
+If the highway is truely connected to the bridge or tunnel, it may only be by a short section of this highway.
+If so, you may have to split the connecting way, and adding bridge or tunnel tags only on the relevant part.'''),
+            trap = T_(
+'''There might be bad detections with connections at the bridge heads or tunnel entrances.'''))
+        self.classs_change[5] = self.def_class(item = 7012, level = 3, tags = ['highway', 'fix:survey', 'fix:imagery'],
+            title = T_('Tunnel connected to non-tunnel highway'),
+            detail = T_(
+'''A bridge or tunnel is usually not connected to regular highways except at the end points.'''),
+            fix = T_(
+'''Disconnect the bridge or tunnel from the highway, or add missing bridge or tunnel tags.
+
+If the highway is truely connected to the bridge or tunnel, it may only be by a short section of this highway.
+If so, you may have to split the connecting way, and adding bridge or tunnel tags only on the relevant part.'''),
+            trap = T_(
+'''There might be bad detections with connections at the bridge heads or tunnel entrances.'''))
 
         self.callback10 = lambda res: {"class":1, "data":[self.way_full, self.positionAsText], "fix":[{"+":{"bridge:structure":"beam"}}, {"+":{"bridge:structure":"suspension"}}] }
         #self.callback20 = lambda res: {"class":2, "data":[self.way_full, self.way_full, self.positionAsText] }
         #self.callback30 = lambda res: {"class":3, "data":[self.way_full, self.positionAsText] }
+        self.callback40 = lambda res: {"class": 4 if res[4] == 'bridge' else 5, "data": [self.node_full, self.way, self.way_full, self.positionAsText] }
```

Comment:
Note to self: forgot to remove `_full` for the non-bridge way, and distinct isn't necessary anymore with the ways in the output